woocommerce/woocommerce-ios
# Login Onboarding: add a survey to the end

Closes: #7405 


### Description


#### Why

In order to understand more about the potential motivation for installing the app, we are adding a survey to the end of the login onboarding as an experiment. Since we don't have an A/B testing setup ready, this feature is enabled for all users in release 9.8 even though I still implemented it under a feature flag.

#### Major changes

The survey view was implemented in SwiftUI in `LoginOnboardingSurveyView`, with a survey option enum `LoginOnboardingSurveyOption`. This survey view is shown above the buttons container in `LoginOnboardingViewController` on top of other pre-existing views as a simple way to implement this experimental feature. I thought about refactoring the existing views but there will be more changes than we want for an experiment. When the user taps "Next" on the survey screen when an option is selected, an event `*_login_onboarding_survey_submitted` is logged with the option. When the survey is shown, we also log an event `*_login_onboarding_survey_shown`.

### Testing instructions


- Reinstall the app if you've seen the login onboarding. Alternatively, you can hardcode the onboarding to be always shown by commenting out [these lines](https://github.com/woocommerce/woocommerce-ios/blob/2c2fc77b9d0611c25694fe9bd96989192c320118/WooCommerce/Classes/ViewRelated/AppCoordinator.swift#L140-L143) in `AppCoordinator`
- Tap "Next" three times --> a survey should be shown with the Next button disabled, and an event `*_login_onboarding_survey_shown` should be logged
- Tap on any option --> the selected option should show the color as in the design, and the Next button becomes enabled
- Tap "Next" to continue to the prologue screen --> an event `*_login_onboarding_survey_submitted` with the corresponding `option` value should be logged

### Screenshots



https://user-images.githubusercontent.com/1945542/182781303-8c63492a-f48b-4a49-ad92-9d15b982b9e2.mov


light | dark
-- | --
![Simulator Screen Shot - iPhone 13 - 2022-08-04 at 12 09 53](https://user-images.githubusercontent.com/1945542/182763458-52a41b96-8f51-4351-8bbf-c61a7ad7b93f.png) | ![Simulator Screen Shot - iPhone 13 - 2022-08-04 at 12 10 00](https://user-images.githubusercontent.com/1945542/182763463-e3761b4d-1023-4967-8285-c7de1623cbed.png)


---
